web/packages/teleterm/src/ui/DocumentGatewayKube/DocumentGatewayKube.tsx

```diff
@@ -84,21 +84,17 @@ export const DocumentGatewayKube = (props: {
       return <DocumentTerminal doc={doc} visible={visible} />;
     }
 
-    case 'error': {
+    default:
       return (
-        <Document visible={visible} px={2}>
-          <Reconnect
-            kubeId={params.kubeId}
-            statusText={connectAttempt.statusText}
+        <Document visible={visible}>
+          <OfflineGateway
```

Comment:
Showing a component that renders something on the screen instead of a blank document causes a problem now.
If you create a kube gateway, then go to resources screen and create it again, you will see a blink of `the kube connection is being set up...` for split of a second. That happens because we send a request to tshd, which resolves almost immediately:
```ts
      await retryWithRelogin(ctx, doc.targetUri, () =>
        // Creating a kube gateway twice with the same params is a noop. tshd
        // will return the URI of an already existing gateway.
        ctx.clustersService.createGateway({
          targetUri: doc.targetUri,
          user: '',
        })
      );
```

This doesn't happen for db gateways because there we check if the gateway exists in the `ClustersService` state before making a request. 
I'm wondering if we could do the same here. This would require adding `gatewayUri` to the kube document and connection. 
@ravicious do you see any problems with changing that?

Reply:
Couldn't we do the same check but with `findGatewayByConnectionParams` instead of `findGateway`? This way we wouldn't need to add `gatewayUri` to the doc.